src/utils/generateArtifacts.js

```diff
@@ -1,41 +1,45 @@
 import parseTargetPath from './parseTargetPath'
 import { generateArtifactPreviewData } from './generateArtifactPreviewData'
 
-import { maxBy } from 'lodash'
+import { maxBy, flatten } from 'lodash'
 
 export const generateArtifacts = artifacts =>
-  artifacts
-    .map(artifact => {
-      const { link_iteration } = artifact.link_iteration ?? {}
-      const generatedArtifact = artifact.link_iteration
-        ? artifact.data.filter(dataItem => dataItem.iter === link_iteration)[0]
-        : maxBy(artifact.data, 'updated')
-
-      if (generatedArtifact) {
-        generatedArtifact.target_path = parseTargetPath(
-          generatedArtifact.target_path
-        )
-
-        if (generatedArtifact.extra_data) {
-          const generatedPreviewData = generateArtifactPreviewData(
-            generatedArtifact.extra_data,
-            generatedArtifact.target_path.schema
-          )
-
-          generatedArtifact.preview = generatedPreviewData.preview
-
-          // @erann March 2, 2021
-          // Not sure why we have this. Commenting out for now.
-          //
-          // if (generatedPreviewData.extraDataPath) {
-          //   generatedArtifact.target_path.path =
-          //     generatedPreviewData.extraDataPath
-          // }
-        } else {
-          generatedArtifact.preview ??= []
+  flatten(
+    artifacts
+      .map(artifact => {
+        const { link_iteration } = artifact.link_iteration ?? {}
+        let generatedArtifacts = artifact.link_iteration
+          ? artifact.data.filter(dataItem => dataItem.iter === link_iteration)
+          : [maxBy(artifact.data, 'updated')]
+
+        if (generatedArtifacts.length > 0) {
+          generatedArtifacts = generatedArtifacts.map(generatedArtifact => {
+            let item = { ...generatedArtifact }
+
+            item.target_path = parseTargetPath(generatedArtifact.target_path)
+
+            if (generatedArtifact.extra_data) {
+              item.preview = generateArtifactPreviewData(
+                generatedArtifact.extra_data,
+                generatedArtifact.target_path.schema
+              )
+
+              // @erann March 2, 2021
+              // Not sure why we have this. Commenting out for now.
+              //
+              // if (generatedPreviewData.extraDataPath) {
+              //   generatedArtifact.target_path.path =
+              //     generatedPreviewData.extraDataPath
+              // }
+            } else {
+              item.preview ??= []
+            }
+
+            return item
+          })
         }
-      }
 
-      return generatedArtifact
-    })
-    .filter(generatedArtifact => !!generatedArtifact)
+        return generatedArtifacts
+      })
+      .filter(generatedArtifact => !!generatedArtifact)
```

Comment:
Let’s be explicit and tell the reader we’re coercing the value of `generatedArtifact` to a Boolean:
```suggestion
      .filter(generatedArtifact => Boolean(generatedArtifact))
```